pymove/visualization/folium.py

```diff
@@ -0,0 +1,2168 @@
+
+    slice_subject_show: int, optional, default
+    """
+
+    # building structure for deltas
+    delta_event = pd.to_timedelta(window_time_event, unit='s')
+    delta_tnz = pd.to_timedelta(window_time_subject, unit='s')
+
+    # length of df_tnz
+    len_df_tnz = df_.shape[0]
+
+    # building structure for lat and lon array
+    lat_arr = np.zeros(len_df_tnz)
+    lon_arr = np.zeros(len_df_tnz)
+
+    # folium map list
+    folium_maps = []
+
+    # for each cvp in df_cvp
+    for _, line in df_event.iterrows():
+
+        event_lat = line[dict_labels['event_lat']]
+
+        event_lon = line[dict_labels['event_lon']]
+
+        event_datetime = line[dict_labels['event_datetime']]
+
+        event_id = line[dict_labels['event_id']]
+
+        # building time window for cvp search
+        start_time = pd.to_datetime(event_datetime - delta_event)
+        end_time = pd.to_datetime(event_datetime + delta_event)
+
+        # filtering df_ for time window
+        df_filted = filters.by_datetime(
+            df_,
+            start_datetime=start_time,
+            end_datetime=end_time
+        )
+
+        # length of df_temp
+        len_df_temp = df_filted.shape[0]
+
+        # using the util part of the array for haversine function
+        lat_arr[:len_df_temp] = event_lat
+        lon_arr[:len_df_temp] = event_lon
+
+        # building distances to cvp column
+        df_filted['distances'] = distances.haversine(
+            lat_arr[:len_df_temp],
+            lon_arr[:len_df_temp],
+            df_filted[dict_labels['tnz_lat']].values,
+            df_filted[dict_labels['tnz_lon']].values
+        )
+
+        # building nearby column
+        df_filted['nearby'] = df_filted['distances'].map(lambda x: (x <= radius))
+
+        # if any data for df_ in cvp time window is True
+        if df_filted['nearby'].any():
+
+            # buildng the df for the first tnz_points of tnz in nearby cvp
+            df_begin = df_filted[df_filted['nearby']].sort_values(
+                dict_labels['tnz_datetime']
+            )
+
+            # making the folium map
+            # folium_map = create_folium_map([df_begin[dict_labels['tnz_lat']].mean(),
+            #                                   df_begin[dict_labels['tnz_lon']].mean()])
+
+            # plot cvp in map
+
+            df_ = df_event[df_event[dict_labels['event_id']] == event_id]
+
+            folium_map = add_event_folium(
+                df_,
+                dict_plot=dict_plot,
+                dict_labels=dict_labels,
+                slice_tags=slice_event_show
+            )
+
+            # keep only the first tnz_point nearby to cvp for each tnz
+            df_begin.drop_duplicates(
+                subset=[dict_labels['tnz_id'], 'nearby'],
+                inplace=True
+            )
+
+            # for each tnz nearby to cvp
+            tnzs = []
+
+            # create an empty dataframe
+            columns_names = list(df_.columns)
```

Comment:
Codacy found an issue: [local variable 'columns_names' is assigned to but never used (F841)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5714604)